apps/main-landing/src/app/creators/donate/content.tsx

```diff
@@ -157,50 +157,44 @@ export const Content = ({ className }: Properties) => {
     'amount',
   ]);
 
-  const allowedTokens = useMemo(() => {
-    const allTokens = Object.values(CHAIN_ID_TO_TOKENS).flat();
-    const uniqueTokens: ChainToken[] = [];
-    for (const token of allTokens) {
-      const exists = uniqueTokens.find((uniqueToken) => {
-        return uniqueToken.symbol === token.symbol;
-      });
-      if (exists) {
-        continue;
-      }
-      uniqueTokens.push(token);
-    }
-
-    return uniqueTokens;
-  }, []);
-
   const sender = useSender({ walletClient, publicClient });
 
   const selectedToken = useMemo(() => {
-    const token = allowedTokens?.find((token) => {
-      return token.address === tokenAddress;
+    const token = possibleTokens?.find((token) => {
+      return token.symbol === tokenAddress;
     });
     setSelectedTokenKey(token?.symbol ?? '');
     return token;
-  }, [allowedTokens, tokenAddress]);
+  }, [possibleTokens, tokenAddress]);
 
   const amountInSelectedToken = useMemo(() => {
-    if (!sender.tokensToSend || !selectedToken?.decimals) {
+    const decimals =
+      CHAIN_ID_TO_TOKENS[chainId]?.find((token) => {
+        return token.symbol === selectedTokenKey;
+      })?.decimals ?? 1;
+    if (!sender.tokensToSend || !selectedToken?.symbol) {
       return;
     }
 
     return applyDecimalsToNumericString(
       sender.tokensToSend.toString(),
-      selectedToken.decimals,
+      decimals,
     );
-  }, [selectedToken?.decimals, sender.tokensToSend]);
+  }, [selectedTokenKey, chainId, sender.tokensToSend]);
 
   const onSubmit: SubmitHandler<SendPayload> = useCallback(
-    async (sendPayload) => {
+    async (payload) => {
       if (!addressValidationResult.success || !validatedAddress) {
         return;
       }
+      const { chainId, tokenAddress: symbol, ...rest } = payload;
+      const address =
```

Comment:
we can add Hex type here to avoid typescript problems:

```
const address: Hex =
        CHAIN_ID_TO_TOKENS[chainId]?.find((token: Token) => {
          return token.symbol === symbol;
        })?.address ?? '0x';